### PR TITLE
Implement Hash#fetch

### PR DESCRIPTION
**Pull request by saveriomiroddi**

### CODE REVIEW

File: vm/hash_test.go [hunk 1]
```diff
@@ -907,49 +946,49 @@ func TestHashMapValuesMethod(t *testing.T) {
 		{`
 		h = { a: 1, b: 2, c: 3 }
 		result = h.map_values do |v|
-		  v * 3
```
Comment (st0012) [inline vm/hash_test.go]: @saveriomiroddi I prefer keeping the two spaces indentation for those Goby code in test cases

File: vm/hash_test.go [hunk 2]
```diff
@@ -623,47 +746,17 @@ func TestHashEqualMethodFail(t *testing.T) {
...
-		`, nil},
-		{`
-		h = { a: 1, b: "Hello", c: true }.delete("a")
-		h["b"]
-		`, "Hello"},
-		{`
-		h = { a: 1, b: "Hello", c: true }.delete("a")
-		h["c"]
-		`, true},
+			{ spaghetti: "eat" }.fetch("pizza", "not eat")
```
Comment (st0012) [inline vm/hash_test.go]: Why don't you test the successfully fetching condition?

Comment (saveriomiroddi) [reply vm/hash_test.go]: You're right, that was an omission :smile:

**Comment by saveriomiroddi** [comment]
@st0012 changes applied; I've entirely removed the commit with the indentation change.

**Pull request merged**